### BC Hydro IPP - Resource Analysis
#### Biogas

BC Hydro acquires power from Independent Power Producers (IPPs) to help meet B.C.'s electricity needs.
IPPs include power production companies, municipalities, First Nations and customers.

The analysis is carried out on the most recent [IPP supply list](https://www.bchydro.com/content/dam/BCHydro/customer-portal/documents/corporate/independent-power-producers-calls-for-power/independent-power-producers/ipp-supply-list-in-operation.pdf) and [IPP supply map](https://www.bchydro.com/content/dam/BCHydro/customer-portal/documents/corporate/independent-power-producers-calls-for-power/independent-power-producers/ipp-supply-map.pdf) obtained from [BC Hydro Website](https://www.bchydro.com/work-with-us/selling-clean-energy/meeting-energy-needs/how-power-is-acquired.html).

In [1]:
import pandas as pd
import altair as alt

In [2]:
data = pd.read_csv('IPP__REGION_list.csv')

#### Initial Description of Data


As of April 1, 2025, BC Hydro has 119 Electricity Purchase Agreements (EPAs) with IPPs whose projects achieved commercial operation, as defined in their EPA and are considered to be in operation. In total, these projects are capable of delivering approximately 18,662 gigawatt hours of annual supply and approximately 5,298 megawatts of capacity.

In [3]:
len(data)

119

In [4]:
IPP_Capacity = round(data['Capacity(MW)'].sum(),2)
IPP_Capacity

5298.84

In [5]:
IPP_Energy = round(data['Energy(GWh/yr)'].sum(),2)
IPP_Energy

18662.3

In [6]:
data.replace({'Sub_region':{'CI':'Central Interior',
                               'LM':'Lower Mainland',
                               'VI':'Vancouver Island',
                               'NC':'North Coast',
                               'EK':'East Kootenay',
                               'PR':'Peace River',
                               'KLN':'Kelly Lake Nicola',
                               'RSAC':'Revelstoke Ashton Creek',
                               'SK':'Selkirk'}}, inplace = True)

In [7]:
peace_list = ['Central Interior','Peace River']

In [8]:
columbia_list = ['Revelstoke Ashton Creek', 'Kelly Lake Nicola', 'Mica', 'Selkirk', 'East Kootenay']

In [9]:
data.loc[data['Sub_region'].isin(columbia_list),'Region'] = 'Columbia Region'
data.loc[data['Sub_region'].isin(peace_list),'Region'] = 'Peace Region'
data.loc[data['Sub_region'] == 'Lower Mainland','Region'] = 'Lower Mainland'
data.loc[data['Sub_region'] == 'North Coast','Region'] = 'North Coast'
data.loc[data['Sub_region'] == 'Vancouver Island','Region'] = 'Vancouver Island'

In [10]:
columns = data.columns.to_list()
for column in columns:
    if data[column].dtype == 'object':
        data[column] = data[column].str.strip()

The capacity and GWhr contribution of various resources by Type is given by the data_capacity_type and data_energy_type respectively 

In [11]:
data_capacity_type = data.groupby('Type').agg({'Capacity(MW)':'sum'}).sort_values(by = 'Capacity(MW)',ascending = False).reset_index()
data_energy_type = data.groupby('Type').agg({'Energy(GWh/yr)':'sum'}).sort_values(by = 'Energy(GWh/yr)',ascending = False).reset_index()

In [12]:
data_energy_type

Type  Energy(GWh/yr)
0  Non-Storage Hydro          7336.4
1      Storage Hydro          4933.3
2            Biomass          2977.6
3               Wind          2161.8
4  Gas Fired Thermal           904.7
5                MSW           166.0
6                ERG           141.0
7             Biogas            37.6
8              Solar             3.9

In [13]:
data_capacity_type

Type  Capacity(MW)
0  Non-Storage Hydro      1983.200
1      Storage Hydro      1292.100
2            Biomass       817.900
3               Wind       747.000
4  Gas Fired Thermal       403.800
5                MSW        24.800
6                ERG        22.800
7             Biogas         5.155
8              Solar         2.090

The type of resource contributing to the largest capacity and energy is Biogas 

#### Waste to Energy 
Waste to Energy Facility generation involves burning fuel to produce hot gases or steam to drive gas or steam turbines that are connected to generators.

Sustainability 2003 Award Winner The tall red and white striped exhaust stack of the GVRD’s Waste-to-Energy Facility (WTEF) in Burnaby is a new flag of sustainability in the GVRD. Built in 1988 to handle over 20% of the region’s solid waste, the facility now generates electricity as well. Over 250,000 tonnes of municipal solid waste are handled at the WTEF each year utilizing three separate processing lines. Built to meet the highest standards of environmental sustainability, it was only the second facility of its kind in North America to qualify for the rigorous ISO 14001 environmental standard. Historically steam from the waste-to-energy facility has been sold to the nearby Norampac paper recycling mill. However, not all of the steam could be utilized by the mill. The plant operators, Montenay Inc., and GVRD engineers saw the excess steam as an opportunity to make the plant more sustainable. They found that generating electricity from the steam and selling the power to BC Hydro provided social, economic and environmental benefits. – Producing electricity from garbage created four new full-time jobs, creating positive social impacts. – Selling the electricity to BC Hydro will generate gross revenues of $5-6 million a year for the GVRD, an economic benefit for the region’s residents. – Reducing the need to generate power elsewhere creates a positive environmental impact. In fact, if BC Hydro had to generate the same power at its Burrard Thermal gas-fired generating station, it would produce 59,000 tonnes of greenhouse gases. Because the project meets social, economic and environmental tests, engineers took the initial letters of the three sustainability tests and called it the SEE-Gen project. Work on the project started in late 2001, using the best available technology. For example, low-noise designs prevent disturbance to wildlife and the surrounding community and an air-cooled condenser reduces the need for cooling water. The SEE-Gen project commenced commercial operation in July 2003. The SEE-Gen project will produce 15 megawatts of electricity, enough to power 15,000 homes. Project engineers are still looking for more savings including ways to generate additional steam and a proposal to sell residual heat from the turbogenerator to a nearby industrial development.

BC Hydro has Electricity Purchase Agreement for Biogas with 66 Independent Power Producers. In total, these 66 projects are capable of delivering approximately 7336.4 gigawatt hours of annual supply and approximately 1983.2 megawatts of capacity. The capacity and GWhr contribution of Biogas in various regions are given by the dataframes energy_Biogas, capacity_Biogas respectively. 

In [14]:
data_Biogas = data[data['Type'] == 'Biogas'].reset_index(drop=True)
energy_Biogas = data_Biogas.groupby('Region').agg({'Energy(GWh/yr)':'sum'}).sort_values(by = 'Energy(GWh/yr)',ascending = False).reset_index()
capacity_Biogas = data_Biogas.groupby('Region').agg({'Capacity(MW)':'sum'}).sort_values(by = 'Capacity(MW)',ascending = False).reset_index()

In [15]:
len(data_Biogas)

2

In [16]:
energy_Biogas

Region  Energy(GWh/yr)
0   Columbia Region            35.6
1  Vancouver Island             2.0

In [17]:
capacity_Biogas

Region  Capacity(MW)
0   Columbia Region         4.800
1  Vancouver Island         0.355

For Biogas, the capacity and energy contributions of each region individually analysed 

#### 1. Vancouver Island


For the Vancouver Island region, the capacity and GWhr contribution of the Biogas resource by location is given in the capacity_VI_Loc and energy_VI_Loc dataframes respectively.

In [18]:
data_VI = data[(data['Type'] == 'Biogas') & (data['Region']=='Vancouver Island')].reset_index(drop=True)
energy_VI_Loc = data_VI.groupby('Location').agg({'Energy(GWh/yr)':'sum'}).sort_values(by = 'Energy(GWh/yr)',ascending = False).reset_index()
capacity_VI_Loc = data_VI.groupby('Location').agg({'Capacity(MW)':'sum'}).sort_values(by = 'Capacity(MW)',ascending = False).reset_index()

In [19]:
energy_VI_Loc

Location  Energy(GWh/yr)
0  Nanaimo             2.0

In [20]:
capacity_VI_Loc

Location  Capacity(MW)
0  Nanaimo         0.355

The bar chart energy_VI_Loc_chart, shows the energy contribution for each of these Biogas projects and the call process by which it was acquired. 

The bar chart capacity_VI_Loc_chart, shows the capacity contribution for each of these Biogas projects and the call process by which it was acquired.


In [21]:
title = alt.TitleParams('The Annual GWhr contribution of Biogas in Vancouver Island by IPP Projects')

capacity_IPP = alt.Chart(data_VI, title = title).mark_bar().encode(
                    alt.Y('Project_name', title='Project Name',sort='x'),
                    alt.X('sum(Energy(GWh/yr))', title='Energy(GWh/yr)'),
                    alt.Color('Call_process', title = 'Call Process')).properties( width=300)
energy_VI_Loc_chart = capacity_IPP + capacity_IPP.mark_text(
    color = 'black', align='left', dx = 3).encode(text='sum(Energy(GWh/yr))',color=alt.value('black'))
energy_VI_Loc_chart

alt.LayerChart(...)

In [22]:
title = alt.TitleParams('The Capacity contribution of Biogas in Vancouver Island by IPP Projects')

capacity_IPP = alt.Chart(data_VI, title = title).mark_bar().encode(
                    alt.Y('Project_name', title='Project Name',sort='x'),
                    alt.X('sum(Capacity(MW))', title='Capacity(MW)'),
                    alt.Color('Call_process', title = 'Call Process')).properties( width=300)
cap_VI_Loc_chart = capacity_IPP + capacity_IPP.mark_text(
    color = 'black', align='left', dx = 3).encode(text='sum(Capacity(MW))',color=alt.value('black'))
cap_VI_Loc_chart

alt.LayerChart(...)

#### 2. Columbia Region

For Columbia region, the capacity and GWhr contribution of the Biogas resource by location is given in the capacity_CR_Loc and energy_CR_Loc dataframes respectively.

In [23]:
data_CR = data[(data['Type'] == 'Biogas') & (data['Region']=='Columbia Region')].reset_index(drop=True)
energy_CR_Loc = data_CR.groupby('Location').agg({'Energy(GWh/yr)':'sum'}).sort_values(by = 'Energy(GWh/yr)',ascending = False).reset_index()
capacity_CR_Loc = data_CR.groupby('Location').agg({'Capacity(MW)':'sum'}).sort_values(by = 'Capacity(MW)',ascending = False).reset_index()

In [24]:
energy_CR_Loc

Location  Energy(GWh/yr)
0  Cache Creek            35.6

In [25]:
capacity_CR_Loc

Location  Capacity(MW)
0  Cache Creek           4.8

The bar chart energy_CR_Loc_chart, shows the energy contribution of the project and the call process by which it was acquired. 

The bar chart capacity_CR_Loc_chart, shows the capacity contribution of each of these projects and the call process by which it was acquired.

In [26]:
title = alt.TitleParams('The Annual GWhr contribution of Biogas in Columbia Region by IPP Project')

capacity_IPP = alt.Chart(data_CR, title = title).mark_bar().encode(
                    alt.Y('Project_name', title='Project Name',sort='x'),
                    alt.X('sum(Energy(GWh/yr))', title='Energy(GWh/yr)'),
                    alt.Color('Call_process', title = 'Call Process')).properties( width=300)
energy_CR_Loc_chart = capacity_IPP + capacity_IPP.mark_text(
    color = 'black', align='left', dx = 3).encode(text='sum(Energy(GWh/yr))',color=alt.value('black'))
energy_CR_Loc_chart

alt.LayerChart(...)

In [27]:
title = alt.TitleParams('The Capacity contribution of Biogas in Columbia Region by IPP Project')

capacity_IPP = alt.Chart(data_CR, title = title).mark_bar().encode(
                    alt.Y('Project_name', title='Project Name',sort='x'),
                    alt.X('sum(Capacity(MW))', title='Capacity(MW)'),
                    alt.Color('Call_process', title = 'Call Process')).properties( width=300)
cap_CR_Loc_chart = capacity_IPP + capacity_IPP.mark_text(
    color = 'black', align='left', dx = 3).encode(text='sum(Capacity(MW))',color=alt.value('black'))
cap_CR_Loc_chart

alt.LayerChart(...)

This study was cariied out to learn about the capacity and energy distribution of Biogas IPP resources around various regions in the province. 